### DISCUSSION TITLE
Cannot connect - authentication required error

### DISCUSSION THREAD

**Discussion by gisornator** [opentelekomcloud/terraform-provider-opentelekomcloud#2062] [Q&A] [answered]
Hello,

I'm trying to use terraform for setting up our infrastructure, however, when I run `terraform plan` I receive the following error:
```

│ Error: failed to authenticate:
│ error generating project client: error extracting token: Authentication Failed, error message: {"error":{"code":401,"message":"The request you have made requires authentication.","title":"Unauthorized"}}
│
│   with provider["registry.terraform.io/opentelekomcloud/opentelekomcloud"],
│   on tmtm_infrastructure.tf line 24, in provider "opentelekomcloud":
│   24: provider "opentelekomcloud" {
```
I attached the config file. Just for clarification: I can use the exact credentials to log in on the console.

[tmtm_infrastructure.txt](https://github.com/opentelekomcloud/terraform-provider-opentelekomcloud/files/10519107/tmtm_infrastructure.txt)

**Comment by anton-sidelnikov** [comment, marked as answer]
@gisornator hi, are you trying to use "https://iam.eu-de.otc.t-systems.com:443/v3" as auth url? Also I think it is better to use clouds.yaml file for auth.
with clouds.yaml provider section will look like this:
```
provider "opentelekomcloud" {
  cloud = "devstack"
}
```

**Reply by gisornator** [reply]
@anton-sidelnikov Hey! Thanks for your answer. Yes I use the auth url you mentioned. 

I use the attached TF file for configuration. I simply followed the documentation on this project page and in the terraform docu: https://registry.terraform.io/providers/opentelekomcloud/opentelekomcloud/latest/docs

This is the configuration for the provider:
```
terraform {
   required_providers {
      opentelekomcloud = {
         source = "opentelekomcloud/opentelekomcloud"
         version = ">= 1.32.3"
      }
   }
}

provider "opentelekomcloud" {
  user_name   = "XXX"
  password    = "XXX"
  domain_name = "XXX"
  tenant_name = "XXX"
  auth_url    = "https://iam.eu-de.otc.t-systems.com/v3"
}
```

**Reply by anton-sidelnikov** [reply]
paste here domain and tenant name please

**Reply by gisornator** [reply]
That should be OTC-EU-DE-00000000001000061336 for both

**Reply by anton-sidelnikov** [reply]
no OTC-EU-DE-00000000001000061336 this is domain name, tenant equal to project name

**Reply by gisornator** [reply]
Apparently it is, at least that's what I got after ordering:
![image](https://user-images.githubusercontent.com/17878740/215413202-e3bb3881-bf9a-42e4-adc3-097e9bfc88b5.png)

Is there a way to figure that out in the console?

**Reply by gisornator** [reply]
Okay, you were right - it was not the tenant name. At least I found the project name under my user settings (eu-de) and that worked.

I got trapped by the wrong tenant name after ordering OTC access.
